includes/class-cron-options-cpt.php

```diff
@@ -0,0 +1,375 @@
+				'orderby'          => 'date',
+				'order'            => 'ASC',
+			) );
+
+			// Nothing more to add
+			if ( empty( $jobs_posts ) ) {
+				break;
+			}
+
+			$page++;
+
+			// Loop through results and built output Core expects
+			if ( ! empty( $jobs_posts ) ) {
+				foreach ( $jobs_posts as $jobs_post ) {
+					$timestamp = strtotime( $jobs_post->post_date_gmt );
+
+					$job_args = maybe_unserialize( $jobs_post->post_content_filtered );
+					if ( ! is_array( $job_args ) ) {
+						continue;
+					}
+
+					$action   = $job_args['action'];
+					$instance = $job_args['instance'];
+					$args     = $job_args['args'];
+
+					$cron_array[ $timestamp ][ $action ][ $instance ] = array(
+						'schedule' => $args['schedule'],
+						'args'     => $args['args'],
+					);
+
+					if ( isset( $args['interval'] ) ) {
+						$cron_array[ $timestamp ][ $action ][ $instance ]['interval'] = $args['interval'];
+					}
+
+				}
+			}
+		} while( true );
+
+		uksort( $cron_array, 'strnatcasecmp' );
+
+		// If we're unscheduling an event, hold onto the previous value so we can identify what's unscheduled
+		if ( $this->is_unscheduling() ) {
+			$this->option_before_unscheduling = $cron_array;
+		} else {
+			$this->option_before_unscheduling = null;
+		}
+
+		return $cron_array;
+	}
+
+	/**
+	 * Handle requests to update the cron option
+	 *
+	 * By returning $old_value, `cron` option won't be updated
+	 */
+	public function update_option( $new_value, $old_value ) {
+		if ( $this->is_unscheduling() ) {
+			$this->unschedule_job( $new_value, $this->option_before_unscheduling );
+		} else {
+			$this->convert_option( $new_value );
+		}
+
+		return $old_value;
+	}
+
+	/**
+	 * Delete jobs that are unscheduled using `wp_unschedule_event()`
+	 */
+	private function unschedule_job( $new_value, $old_value ) {
+		$jobs = $this->find_unscheduled_jobs( $new_value, $old_value );
+
+		foreach ( $jobs as $job ) {
+			$this->delete_job( $job['timestamp'], $job['action'], $job['instance'] );
+		}
+	}
+
+	/**
+	 * Save cron events in CPT
+	 */
+	private function convert_option( $new_value ) {
+		if ( is_array( $new_value ) && ! empty( $new_value ) ) {
+			$events = collapse_events_array( $new_value );
+
+			foreach ( $events as $event ) {
+				$job_exists = $this->job_exists( array(
+					'name'             => $this->event_name( $event['timestamp'], $event['action'], $event['instance'] ),
+					'post_type'        => self::POST_TYPE,
+					'post_status'      => self::POST_STATUS,
+					'suppress_filters' => false,
+					'posts_per_page'   => 1,
+				) );
+
+				if ( ! $job_exists ) {
+					// Build minimum information needed to create a post
+					$job_post = array(
+						'post_title'            => $this->event_title( $event['timestamp'], $event['action'], $event['instance'] ),
+						'post_name'             => $this->event_name( $event['timestamp'], $event['action'], $event['instance'] ),
+						'post_content_filtered' => maybe_serialize( array(
```

Comment:
Since we're only using `post_content_filtered`, there's no risk. Neither `wp_insert_post()` nor `sanitize_post_field()` modify the value from that column.